src/jit/codegencommon.cpp

```diff
@@ -10824,6 +10824,9 @@ CORINFO_CLASS_HANDLE Compiler::GetHfaClassHandle(GenTreePtr tree)
         case GT_ASG:
             assert(tree->gtOp.gtOp1->gtOper == GT_LCL_VAR || tree->gtOp.gtOp1->gtOper == GT_LCL_FLD);
             return GetHfaClassHandle(tree->gtOp.gtOp1);
+            
+        default:
+            break;
```

Comment:
That's correct, clang throws a warning on cases for enums when they lack cases for all values. I can disable the warning and revert these changes if you'd like to keep this purely functional, as above with the commenting after #ifdef.